### PR TITLE
added `Should (Not) Contain Any` keywords #2120

### PR DESCRIPTION
**Pull request by jimni**

### CODE REVIEW

File: src/robot/libraries/BuiltIn.py [hunk 1]
```diff
@@ -838,6 +838,62 @@ def should_contain(self, container, item, msg=None, values=True):
...
+
+        Note that possible equal signs in ``*items`` must be escaped
+        with a backslash (e.g. ``name\=value``) to avoid them to be passed in
+        as ``**configuration``.
+
+        Examples:
+        | Should Contain Any | ${some list} | item 1 | item 2 |
+        | Should Contain Any | ${string} | substring | letter | punctuation! |
+        | Should Contain Any | ${container} | item 1 | ... | item n | msg=Keyword Failed | values=False |
+        """
```
Comment (pekkaklarck) [inline src/robot/libraries/BuiltIn.py]: This kind of note needed:

> New in Robot Framework 3.0.1.

File: src/robot/libraries/BuiltIn.py [hunk 2]
```diff
@@ -838,6 +838,62 @@ def should_contain(self, container, item, msg=None, values=True):
...
+        msg = configuration.pop('msg', None)
+        values = configuration.pop('values', True)
+        if configuration:
+            raise RuntimeError("Unsupported configuration parameters were used: %s."
+                               % unic(seq2str(configuration.keys())))
```
Comment (pekkaklarck) [inline src/robot/libraries/BuiltIn.py]: Both `unic` and `.keys()` are unnecessary.

File: src/robot/libraries/BuiltIn.py [hunk 3]
```diff
@@ -838,6 +838,62 @@ def should_contain(self, container, item, msg=None, values=True):
...
+
```
Comment (pekkaklarck) [inline src/robot/libraries/BuiltIn.py]: We prefer not to have empty lines inside functions or methods. #nitpick

File: atest/testdata/standard_libraries/builtin/verify.robot [hunk 4]
```diff
@@ -293,6 +297,44 @@ Should Contain With Non-String Values
     Should Contain    ${dict}    a
     Should Contain    ${dict}    c
 
+Should Contain Any
+    [Documentation]    FAIL Message: 'abcdefg' does not contain any of ''A', 'X' or 'Ф''
```
Comment (pekkaklarck) [inline atest/testdata/standard_libraries/builtin/verify.robot]: There are extra quotes around the list of items but that's due to the `_get_string_msg` helper unconditionally adding them. I guess the helper could get a new config parameters like `quote_str1=True` and `quote_str2=True` to control that, and the new keywords could call it like `quote_str2=False`. What do you think?

Somewhat related to this, I'd prefer the `_get_string_msg` to add a full stop at the end of the default message. That's totally unrelated to this PR (the change require changing lot of tests) and, unless you want to help with that too, I can take care of that myself.

**Comment by pekkaklarck** [comment]
Looks very good. I added some line notes about things I think should still be looked at. In addition to that, please add yourself to AUTHORS.txt file.

**Comment by pekkaklarck** [comment]
See also our contribution guidelines: https://github.com/robotframework/robotframework/blob/master/CONTRIBUTING.rst

**Comment by jimni** [comment]
Thank you very much for your feedback. I will have time later this week to look into all your comments and modify my fix accordingly. I hope such delay is acceptable for this issue.

**Comment by pekkaklarck** [comment]
No worries. I've been busy with other tasks myself and haven't had time to concentrate on 3.0.1 too much either.

**Comment by jimni** [comment]
I believe I have fixed everything you mentioned in the line notes. As per your helper suggestion, I started another issue #2403, please take a look when it's convenient for you.

Is there anything else you would like me to add to the current PR?

**Comment by pekkaklarck** [comment]
The PR looks very good. I could merge it right away, but it would be a bit better if `_get_string_msg` was enhanced to allow dropping extra quotes first. Squashing the commits when everything is done would be nice too.

**Comment by jimni** [comment]
You want `_get_string_msg` fix to be in this branch, correct?

**Comment by pekkaklarck** [comment]
Depends on the changes to `_get_string_msg`. If you only enhance it to allow dropping quotes, that can be part of this PR. Adding `.` to the end should be a separate PR because it is totally unrelated to this PR and requires updating so many tests. That `.` is *very* low priority anyway.

**Comment by jimni** [comment]
Ready to squash once you accepted the changes to `_get_string_message`. 
Please let me know if the code needs anything else or if my fix doesn't comply with the project's coding practice.

**Comment by jimni** [comment]
Will be mostly unable to contribute for the next couple weeks, sorry for the possible inconvenience.

**Comment by pekkaklarck** [comment]
Been busy with holidays, house moves,and lot of training courses, but now back working with RF 3.0.1. This PR seems to be in a really good shape and ready to be merged. The only issue I saw is that the error message when there are no `*items` is likely not too good. I'll check how it looks in practice and possibly improve it myself.

**Comment by jimni** [comment]
Thanks a lot for all your input. It's been a pleasure working on this fix with you.

**Pull request merged**